### PR TITLE
Add zstd pre-and-postprocessor.

### LINKED ISSUE #1100: Add `zstd` pre- and postprocessors
**Describe the problem you are trying to solve**

Lemire has been publishing a small measurement about zstd vs. gzip for json compression https://lemire.me/blog/2021/06/30/compressing-json-gzip-vs-zstd/ . And zstd is super fast and offers better compression. But we dont have zstd yet.

**Describe the solution you'd like**

We want what lemire has: `zstd`.
We want `zstd` pre- and postprocessors similar to what we have for `zlib` etc.

### PR DESCRIPTION
**Pull request by dak-x**
Signed-off-by: Daksh Chauhan <dak-x@outlook.com>

# Pull request
Add `zstd` pre-and-postprocessors

## Related

* Related Issues: closes #1100 

## Checklist

* [X] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment
* [X] Update CHANGELOG.md appropriately, recording any changes, bug fixes, or other observable changes in behavior
* [x] The performance impact of the change is measured (see below)

### CODE REVIEW

**Comment by dak-x** [comment]
The code required seems too straightforward. I feel I am missing some stuff. Also I should probably add some tests, but I do not know what tests.

File: src/preprocessor.rs [hunk 1]
```diff
@@ -69,6 +69,7 @@ pub fn lookup(name: &str) -> Result<Box<dyn Preprocessor>> {
         "ingest-ns" => Ok(Box::new(ExtractIngresTs {})),
         "length-prefixed" => Ok(Box::new(LengthPrefix::default())),
         "textual-length-prefix" => Ok(Box::new(TextualLength::default())),
+        "zstd" => Ok(Box::new(Zstd::default())),
```
Comment (Licenser) [inline src/preprocessor.rs:72]: We should include this in the `decompress` preprocessor for autodetecting zstd compressed data.

**Review by Licenser — CHANGES_REQUESTED** [review]
Heya :) awesome! thank you! only a few small things, one I commented inline, then there should be some tests for the processors and the lookup. You can find examples on how to dot hat in the `preprocessors.rs` there the other compression algorithms are tested too.

The https://github.com/tremor-rs/tremor-www-docs/blob/main/docs/artefacts/postprocessors.md and https://github.com/tremor-rs/tremor-www-docs/blob/main/docs/artefacts/preprocessors.md will need updating to reflect the new processor.

Otherwise 👍 great job :D

File: src/preprocessor.rs (hunk 1 above)
Comment (darach) [reply src/preprocessor.rs:72]: LGTM - Adding fingerprinting to `decompress` is the last piece of the puzzle - it's very useful when the decompression algorithm can vary. Great work!

**Comment by mfelsche** [comment]
That was quick! 😎 🚀

**Comment by mfelsche** [comment]
We might also need to add libzstd as dependencies to our rpm, deb packages and to the docker image.

**Comment by dak-x** [comment]
@mfelsche Where should I update this dependency for the build.

**Comment by mfelsche** [comment]
I think zstd brings zstd-sys which is bringing its own `libzstd`, so it is actually fine, no need to add anything :) sorry for the confusion.

**Review by mfelsche — APPROVED** [review]

**Review by Licenser — APPROVED** [review]
Woooh thank you!! 🚀

**Pull request merged**